src/net/http/client.go

```diff
@@ -59,7 +59,7 @@ type Client struct {
 	// Transport specifies the mechanism by which individual
 	// HTTP requests are made.
 	// If nil, DefaultTransport is used.
-	Transport RoundTripper
+	Transport ExtendedRoundTripper
```

Comment:
This is a public field. This would break any caller who doesn't pass in a RoundTripper without `RoundTripWithCallback`.

Reply:
What is the problem you want to solve?

Reply:
That's a fair point. We can just extend the struct rather than modify this field. That's easy enough.